### DISCUSSION TITLE
having trouble playing an animation only on keydown or other trigger

### DISCUSSION THREAD

**Discussion by michaelgenecin95** [bonsai-rx/bonsai#1542] [Q&A] [answered]
Hi all,
I am trying to set up a looming stimulus animation, just a circle that grows in diameter. However, I want to present the blank white screen at all times until there is a trigger (in this case, key down), at which point the stimulus appears and grows in diameter over a 0.5s period. I'm able to draw the animation, and control when it repeats, but I can't figure out how to clear the circle from the window at the end. I think there may be something obvious I am missing... 

I have included a jpg of my workflow below. Here is what happens in it's current form: On initiating, the animation occurs once (a small dot appears and grows in diameter up to a diameter of 1). The large circle with diameter of 1 stays in the window indefinitely, until I press key down, at which point the animation replays. 

![bonsai_issue_animation](https://github.com/bonsai-rx/bonsai/assets/55214816/e23465fb-650c-448e-b1b2-3c558e4946cd)

[LoomingExample_animation.zip](https://github.com/bonsai-rx/bonsai/files/12489871/LoomingExample_animation.zip)

**Comment by bruno-f-cruz** [comment]
Please add the workflow as suggested in #864 
Thanks!

**Comment by michaelgenecin95** [comment]
added it above, thanks!

**Reply by bruno-f-cruz** [reply, marked as answer]
Switch a takeUntil to a subscribeWhen, and use the end of the animation to close the sequence and re-arm the trigger.
E.g.:
![image](https://github.com/bonsai-rx/bonsai/assets/7049351/7411c325-cc47-4d79-9875-ea4a67833710)

**Reply by michaelgenecin95** [reply]
works great, thanks so much!!
